### ISSUE TITLE
System.InvalidOperationException: 'No browser configured.' 

### ISSUE THREAD

**Issue by asusralis** [auth0/auth0-oidc-client-net#118]
I was following [this](https://auth0.com/blog/developing-mobile-apps-with-xamarin-forms-and-azure-functions/) guide to having auth0 authentication in Xamarin.Forms, but I ran across an exception that I can't find any information about: 

  at IdentityModel.OidcClient.AuthorizeClient.AuthorizeAsync (IdentityModel.OidcClient.Browser.DisplayMode displayMode, System.Int32 timeout, System.Object extraParameters) [0x00036] in C:\local\identity\model\IdentityModel.OidcClient\src\AuthorizeClient.cs:36 
  at IdentityModel.OidcClient.OidcClient.LoginAsync (IdentityModel.OidcClient.LoginRequest request) [0x000ee] in C:\local\identity\model\IdentityModel.OidcClient\src\OidcClient.cs:94 

```
var auth0LoginResult = await _auth0Client.LoginAsync(new { audience = Auth0Config.Audience });
```

This happens on both emulator and device in Android. I set up the intent filter on my activity. I was using Azure AD B2C login on the same device, so I don't think it has to do with my browser. I see [here](https://github.com/IdentityModel/IdentityModel.OidcClient/blob/master/src/AuthorizeClient.cs) where the exception is, but the guide I was following didn't have to set the browser property.

Auth0.Core - 6.5.4
Auth0.Android/iOS - 3.0.0
Xamarin.Forms - 4.3.0.947036

**Comment by damieng** [comment]
Hi.  That code should still work in 3.0.0 but it looks like there's a bug. 

We'll also need to update that blog post for 3.0 as there are some new features including one that will get you immediately past this problem without waiting for a patch.

Instead of that AuthenticationService create the client in your main activity and pass "this" in, e.g.

```csharp
_auth0Client = new Auth0Client(new Auth0ClientOptions
{
    Domain = AuthenticationConfig.Domain,
    ClientId = AuthenticationConfig.ClientId
}, this);
```

**Comment by asusralis** [comment]
Hey, when passing in the main activity I get this:

System.TypeLoadException
  Message=Parent class vtable failed to initialize, due to: VTable setup of type Auth0.OidcClient.ChromeCustomTabsBrowser failed assembly:/storage/emulated/0/Android/data/x/files/.__override__/Auth0.OidcClient.dll type:ChromeCustomTabsBrowser member:(null)

```
            var _auth0Client = new Auth0.OidcClient.Auth0Client(new Auth0.OidcClient.Auth0ClientOptions
            {
                Domain = Auth0Config.Domain,
                ClientId = Auth0Config.ClientId
            }, this);
```

**Comment by damieng** [comment]
Can you make sure you project has a reference to 'Auth0.OidcClient.Core' as well?

**Comment by asusralis** [comment]
I only had Auth0.Core installed, but adding that fixed it. I think that package is missing [here?](https://auth0.com/docs/quickstart/native/xamarin/01-login)

**Comment by damieng** [comment]
It should automatically be brought in - I think I know why it wasn't and that will be going out with the other fix in 3.0.1 shortly.

**Comment by damieng** [comment]
3.0.1 is out now which should fix these two issues.
